service/authentication_validation.go

```diff
@@ -32,6 +33,10 @@ func ValidateAuthenticationCreationRequest(auth *model.AuthenticationCreateReque
 	// capitalize it so it's always the same format.
 	auth.ResourceType = util.Capitalize(auth.ResourceType)
 
+	if err = ValidateAzureSubscriptionId(auth); err != nil {
+		return fmt.Errorf("subscription ID is invalid")
```

Comment:
Here the `err` is getting discarded, because you're creating a new `Errorf` but you're not wrapping the previous error.

```suggestion
		return fmt.Errorf("subscription ID is invalid: %w", err)
```